main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/gokrazy/gokrazy"
+)
+
+var (
+	updateEndpoint string
+	checkInterval  string
+)
+
+func main() {
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	log.Print("gokrazy's selfupdate service starting up..")
+
+	flag.StringVar(&updateEndpoint, "update_endpoint", "", "the HTTP/S endpoint of the update service (required)")
+	flag.StringVar(&checkInterval, "check_interval", "1h", "the time duration interval between checks to the update service. default: 1h")
+
+	flag.Parse()
+
+	if updateEndpoint == "" {
+		log.Fatalln("flag --update_endpoint must be provided")
+	}
+
+	interval, err := time.ParseDuration(checkInterval)
+	if err != nil {
+		log.Fatalln(fmt.Errorf("failed to parse check-interval duration: %w", err))
+	}
+
+	machineID := gokrazy.MachineID()
+
+	_, sbomHash, err := gokrazy.ReadSBOM()
+	if err != nil {
+		log.Fatalf("could not read SBOM from disk: %s", err.Error())
+	}
+
+	httpPassword, err := readConfigFile("gokr-pw.txt")
+	if err != nil {
+		log.Fatalf("could read neither /perm/gokr-pw.txt, nor /etc/gokr-pw.txt, nor /gokr-pw.txt: %s", err.Error())
+	}
+
+	httpPort, err := readConfigFile("http-port.txt")
+	if err != nil {
+		log.Fatalf("could read neither /perm/http-port.txt, nor /etc/http-port.txt, nor /http-port.txt: %s", err.Error())
+	}
+
+	log.Printf("entering update checking loop with interval: %s", interval.String())
+	ticker := time.NewTicker(interval)
```

Comment:
Can you add jitter in the range of a few seconds to minutes? https://github.com/gokrazy/gokrazy/blob/9cdaabefeec8cd194f0a14409f0b85f1c6217540/cmd/heartbeat/heartbeat.go#L133